src/kernel/pci.c

```diff
@@ -280,6 +381,22 @@ pci_probe_bus(int bus)
     }
 }
 
+void pci_bus_set_iomem(int bus, id_heap iomem)
+{
+    pci_bus pbus = allocate(pci_buses->h, sizeof(*pbus));
+    assert(pbus != INVALID_ADDRESS);
+    pbus->iomem = iomem;
+    assert(vector_set(pci_buses, bus, pbus));
```

Comment:
Just to confirm: Does pci_buses does not need to be locked because it is only ever modified on init (scanning of PCI root bridges)?

Reply:
Correct, pci_buses is only ever modified when init_acpi_tables() is called during boot